utils/auth/auth.controller.js

```diff
@@ -0,0 +1,55 @@
+const bcrypt = require("bcryptjs");
+const validate = require("../../resources/users/users.validation");
+const Auth = require("./auth");
+const { genSaltSync, hashSync } = bcrypt;
+
+module.exports = function(model) {
+  return {
+    async createUser(req, res) {
+      const { error } = validate(req.body);
+
+      if (error) {
+        return res
+          .status(422)
+          .json({ status: 422, error: error.details[0].message });
+      }
+      const { first_name, last_name, username, email, password } = req.body;
+      let doc = await model.findOne({ email: req.body.email });
+      if (doc) {
+        return res.status(400).send("That user already exisits!");
+      } else {
+        // Insert the new user if they do not exist yet
+        const salt = genSaltSync(10);
+        const hash = hashSync(password, salt);
+        doc = new model({
+          first_name,
+          last_name,
+          username,
+          email,
+          password: hash
+        });
+        const payload = {
+          id: doc.id,
+          username: doc.username
+        };
+
+        const options = {
+          expiresIn: "24h"
+        };
+
+        const token = Auth(payload, options);
```

Comment:
The `Auth()` is confusing may be you should rename it to be `createToken` , `generateToken`  or `Auth.generateToken`